core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -17,7 +17,7 @@
 
 package org.apache.spark.ui.jobs
 
-import scala.collection.mutable.{HashMap, HashSet, ListBuffer}
+import scala.collection.mutable.{HashMap, HashSet, ListBuffer, LinkedHashMap}
```

Comment:
if it's safe to do `remove` on a HashMap, I think it's fine to revert this back to using a HashMap. It's more consistent in that case.